go/runtime/host/sgx/tcb_cache.go

```diff
@@ -0,0 +1,119 @@
+package sgx
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/common/logging"
+	"github.com/oasisprotocol/oasis-core/go/common/persistent"
+	"github.com/oasisprotocol/oasis-core/go/common/sgx/pcs"
+)
+
+const (
+	tcbCacheKey = "tcb_bundle_cache"
+
+	tcbCacheRefreshThreshold    = 14 * 24 * time.Hour
+	tcbCacheSlowRefreshInterval = 24 * time.Hour
+)
+
+type tcbBundleCache struct {
+	Bundle         *pcs.TCBBundle `json:"bundle"`
+	ExpectedExpiry time.Time      `json:"expected_expiry"`
+	LastUpdate     time.Time      `json:"last_update"`
+}
+
+type tcbCache struct {
+	serviceStore *persistent.ServiceStore
+	logger       *logging.Logger
+	now          func() time.Time
+}
+
+func (tc *tcbCache) check() (bundle *pcs.TCBBundle, refresh bool) {
+	var err error
+
+	// Check if we have a copy in the local store.
+	var stored *tcbBundleCache
+	switch err = tc.serviceStore.GetCBOR([]byte(tcbCacheKey), &stored); err {
+	case nil, persistent.ErrNotFound:
+		// Not cached yet. Not an error.
+	default:
+		// Can't get it... an error, but we can still try downloading it.
+		if tc.logger != nil {
```

Comment:
Or also create the logger in the `mockTbcCache`. I think logging an error here makes sense, since this is an internal to cache error, that's intended for the node operator not for the calling code.